### PR TITLE
Fix NPE for transform aggregations

### PR DESCRIPTION
**Pull request by Hailong-am**
*Issue #, if available:*
#829 

*Description of changes:*

*CheckList:*
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/index-management/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Hailong-am** [comment]
may need to merge after https://github.com/opensearch-project/index-management/pull/825

File: src/main/kotlin/org/opensearch/indexmanagement/transform/model/Transform.kt [hunk 1]
```diff
@@ -363,7 +363,10 @@ data class Transform(
                             groups.add(Dimension.parse(xcp))
                         }
                     }
-                    AGGREGATIONS_FIELD -> aggregations = AggregatorFactories.parseAggregators(xcp)
+                    AGGREGATIONS_FIELD -> {
+                        aggregations =
```
Comment (zhichao-aws) [inline src/main/kotlin/org/opensearch/indexmanagement/transform/model/Transform.kt:367]: can use this expression:
AGGREGATIONS_FIELD -> AggregatorFactories.parseAggregators(xcp)?.let { aggregations = it }

**Comment by zhichao-aws** [comment]
caused by https://github.com/opensearch-project/OpenSearch/pull/8206

File: src/main/kotlin/org/opensearch/indexmanagement/transform/model/Transform.kt (hunk 1 above)
Comment (Hailong-am) [reply src/main/kotlin/org/opensearch/indexmanagement/transform/model/Transform.kt]: sounds good.

**Review by zhichao-aws — APPROVED** [review]
lgtm

**Review by gaobinlong — APPROVED** [review]

**Review by xluo-aws — APPROVED** [review]

**Pull request merged**